uhc: Use Stackage LTS 4.1 to build UHC

phile314 opened the pull request:
UHC release cycle doesn't align nicely with Hackage's constant stream of new package releases. To make maintenance of the Nix UHC package easier, the Stackage LTS package set should be used.

PR tested on Linux Fedora 64bit.

Maybe the UHC build needs to be disabled on Hydra, as it uses Stackage LTS now ?

phile314 committed:
uhc: Use Stackage LTS 4.1 to build UHC

phile314 commented:
See also https://github.com/UU-ComputerScience/uhc/issues/69

peti commented:
Yes, hydra builds for UHC would definitely have to be disabled, because Hydra would end up compiling many, many packages just for this build. I think that would be too expensive.

peti commented:
Also, I'd very much prefer to use LTS 5.x for this, if possible.

phile314 commented:
I agree that LTS 5.0 would be better. The problematic package is hashable 1.2.4.0 or newer. The latest LTS with an old enough version of hashable is LTS 4.1, so there is not much choice there.

With this PR/change, the UHC build might lag a bit behind current Stackage/Hackage versions, but I think this is better than having broken UHC builds half of the time.

I updated the PR, added some more comments and disabled hydra builds.
